src/backend/marsha/core/api/video.py

```diff
@@ -820,3 +821,105 @@ def initiate_upload(self, request, pk=None):
         Thumbnail.objects.filter(pk=pk).update(upload_state=defaults.PENDING)
 
         return Response(presigned_post)
+
+
+class SharedLiveMediaViewSet(ObjectPkMixin, viewsets.ModelViewSet):
+    """Viewset for the API of the SharedLiveMedia object."""
+
+    permission_classes = [permissions.NotAllowed]
+    queryset = SharedLiveMedia.objects.all()
+    serializer_class = serializers.SharedLiveMediaSerializer
+
+    def get_permissions(self):
+        """Instantiate and return the list of permissions that this view requires."""
+        if self.action in ["create", "list"]:
+            permission_classes = [
+                permissions.IsTokenInstructor
+                | permissions.IsTokenAdmin
+                | permissions.IsParamsVideoAdminThroughOrganization
+                | permissions.IsParamsVideoAdminThroughPlaylist
+            ]
+        elif self.action in ["retrieve"]:
+            permission_classes = [
+                permissions.IsTokenResourceRouteObjectRelatedVideo
+                | permissions.IsRelatedVideoPlaylistAdmin
```

Comment:
I'm wondering if this is covering instructors?

Reply:
No, only administrator of the playlist. At took the same permissions from `timed text tracks`. We can later open a little bit more but for now I suggest to not do it while we are not testing it in a real context.